### PR TITLE
fix: Nx 16 package module types

### PR DESCRIPTION
**Pull request by maroun9t**
- Add missing type fields in packages
- Add a new project containing tests for the generated package.json

### CODE REVIEW

File: packages/tests/generated-packages/project.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      "executor": "@nx/linter:eslint",
+      "outputs": ["{options.outputFile}"],
+      "options": {
+        "lintFilePatterns": ["packages/tests/generated-packages/**/*.ts"]
+      }
+    },
+    "test": {
+      "executor": "@nx/jest:jest",
+      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
+      "dependsOn": ["build-all-no-playgrounds"],
```
Comment (maroun9t) [inline packages/tests/generated-packages/project.json:17]: @Lp-Francois How can I make this `dependsOn` actually work? 
Currently it doesn't seem to invoke the `build-all-no-playgrounds` target.

Comment (Lp-Francois) [reply packages/tests/generated-packages/project.json:17]: Answered on Slack

File: packages/plugins/privacy/package.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "@ninetailed/experience.js-plugin-privacy",
   "version": "7.6.0-beta.1",
+  "type": "commonjs",
```
Comment (maxrawlinger) [inline packages/plugins/privacy/package.json:4]: Why do we use commonjs here while all other packages are type module?

Comment (maroun9t) [reply packages/plugins/privacy/package.json:4]: It's actually not the only one. 

NX@16 is no longer adding the `type` field to all the generated packages. 
To enforce having the `type` field in all packages we add it manually based on these two criteria:

- We either match what we have in the generated packages prior to the NX@16 upgrade.
- If the package doesn't have an `esm`  build, we give it a type of `commonjs`.

**Pull request merged**